packages/vm/src/runTx.ts

```diff
@@ -546,22 +544,12 @@ export async function generateTxReceipt(
         ...baseReceipt,
       } as PreByzantiumTxReceipt
     }
-  } else if ('transactionType' in tx && tx.transactionType === 1) {
-    // EIP2930 Transaction
-    receipt = {
-      status: txResult.execResult.exceptionError ? 0 : 1,
-      ...baseReceipt,
-    } as EIP2930Receipt
-  } else if ('transactionType' in tx && tx.transactionType === 2) {
-    // EIP1559 Transaction
+  } else {
+    // Typed EIP-2718 Transaction
     receipt = {
       status: txResult.execResult.exceptionError ? 0 : 1,
       ...baseReceipt,
-    } as EIP1559Receipt
-  } else {
-    throw new Error(
-      `Unsupported transaction type ${'transactionType' in tx ? tx.transactionType : 'NaN'}`
-    )
+    } as PostByzantiumTxReceipt
```

Comment:
I have simplified here since I think this is not very future proof if we bring out some new tx type within a new non-breaking tx release and this would be run on a new VM (this would then break here without the change) and at the same time I am not really seeing the use for these dedicated receipt types.

Hope that I am not overlooking some side effects here though, eventually also give this some additional thought if you think this is ok.

Reply:
(and *if* there will be some future tx type changing the receipt format once again we can still *then* think about how we react and eventually add a specific new receipt type here)

Reply:
I believe the reason this was implemented with dedicated receipt names for each EIP is due to EIP-2718 saying:

> * `Receipt` is either `TransactionType || ReceiptPayload` or `LegacyReceipt`
> * `ReceiptPayload` is an opaque byte array whose interpretation is dependent on the `TransactionType` and defined in future EIPs
> * `LegacyReceipt` is `rlp([status, cumulativeGasUsed, logsBloom, logs])`

EIP-2930:

> The EIP-2718 `ReceiptPayload` for this transaction is `rlp([status, cumulativeGasUsed, logsBloom, logs])`.

EIP-1559:

> The EIP-2718 `ReceiptPayload` for this transaction is `rlp([status, cumulativeGasUsed, logsBloom, logs])`.

So although the receipts are aligned so far, they might deviate in the future. That said, I think your code improvements and reasoning here are great and we can introduce a new receipt type when the time comes.